src/ArticleCard.jsx

```diff
@@ -15,6 +16,23 @@ function ArticleCard() {
       setIsLoading(false);
     });
   }, [article_id]);
+
+  useEffect(() => {
```

Comment:
This function doesn't need to be in a useEffect. They aren't tied to the components rendering, but to user behaviour (i.e. clicking on the button). At the moment, it is running when you refresh the page, but not when a user clicks on the button.